src/devices/cpu/arcompact/arcompactdasm.h

```diff
@@ -12,6 +12,84 @@
 
 #pragma once
 
+#define DASM_GET_01_01_01_BRANCH_ADDR \
+	int32_t address = (op & 0x00fe0000) >> 17; \
+	address |= ((op & 0x00008000) >> 15) << 7; \
+	if (address & 0x80) address = -0x80 + (address & 0x7f); \
+	op &= ~ 0x00fe800f;
+
+
+#define DASM_GROUP_0e_GET_h \
+	h =  ((op & 0x0007) << 3); \
+	h |= ((op & 0x00e0) >> 5); \
+	op &= ~0x00e7;
+#define DASM_COMMON32_GET_breg \
+	int b_temp = (op & 0x07000000) >> 24; op &= ~0x07000000; \
+	int B_temp = (op & 0x00007000) >> 12; op &= ~0x00007000; \
+	int breg = b_temp | (B_temp << 3);
+#define DASM_COMMON32_GET_creg \
+	int creg = (op & 0x00000fc0) >> 6; op &= ~0x00000fc0;
+#define DASM_COMMON32_GET_u6 \
+	int u = (op & 0x00000fc0) >> 6; op &= ~0x00000fc0;
+#define DASM_COMMON32_GET_areg \
+	int areg = (op & 0x0000003f) >> 0; op &= ~0x0000003f;
+#define DASM_COMMON32_GET_areg_reserved \
+	int ares = (op & 0x0000003f) >> 0; op &= ~0x0000003f;
+#define DASM_COMMON32_GET_F \
+	int F = (op & 0x00008000) >> 15; op &= ~0x00008000;
+#define DASM_COMMON32_GET_p \
+	int p = (op & 0x00c00000) >> 22; op &= ~0x00c00000;
+
+#define DASM_COMMON32_GET_s12 \
+		int S_temp = (op & 0x0000003f) >> 0; op &= ~0x0000003f; \
+		int s_temp = (op & 0x00000fc0) >> 6; op &= ~0x00000fc0; \
+		int S = s_temp | (S_temp<<6);
+#define DASM_COMMON32_GET_CONDITION \
+		uint8_t condition = op & 0x0000001f;  op &= ~0x0000001f;
```

Comment:
These macros will also need to go in a local .ipp file so they don’t leak downstream.

Reply:
yeah, as the initial version didn't get reviewed/merged I dropped it down to draft and will just start the refactoring now.  There's a good chance a lot of these will simply go away anyway, or become functions.